src/components/tests/tour.test.tsx

```diff
@@ -0,0 +1,223 @@
+import { createMount } from '@material-ui/core/test-utils';
+import React from 'react';
+import DashboardTour from '../tour';
+import { SpotlightValue } from '../spotlight-value';
+import { ReactWrapper } from 'enzyme';
+import { OmniPanelViewIndex } from '../dashboard';
+
+/*react-leaflet does not work well on jsdom,
+suppress warnings of not being able to find elements from leaflet*/
+jest.spyOn(console, 'warn').mockImplementation(() => {});
+
+const mount = createMount();
+
+describe('Dashboard Tour', () => {
+  let tourState: boolean = true;
+  let setTourState: jest.Mock = jest.fn();
+  let setTourShowOmniPanel: jest.Mock = jest.fn();
+  let setTourSettingsAndOmniPanel: jest.Mock = jest.fn();
+  let doorSpotlight: SpotlightValue<string> = { value: 'main_door' };
+  let setDoorSpotlight: jest.Mock = jest.fn();
+
+  let root: ReactWrapper;
+  let expectedText: string;
+  let stepText: string;
+
+  beforeEach(() => {
+    const tourProps = {
+      tourState,
+      setTourState,
+      setTourShowOmniPanel,
+      setTourSettingsAndOmniPanel,
+      OmniPanelViewIndex,
+      doorSpotlight,
+      setDoorSpotlight,
+    };
+    root = mount(<DashboardTour tourProps={tourProps} />);
+  });
+
+  afterEach(() => {
+    root?.unmount();
+  });
+
+  const wait = (delay = 0) => new Promise((resolve) => setTimeout(resolve, delay));
+
+  it('renders without crashing', () => {
+    expect(root.html()).toMatchSnapshot();
+  });
+
+  it('renders first welcome step (1/12)', () => {
+    expectedText = 'Welcome to RoMi dashboard';
+    stepText = root.find('.reactour__helper').hostNodes().text();
+
+    expect(stepText.includes(expectedText));
+    expect(root.html()).toMatchSnapshot();
+  });
+
+  it('renders zoom button step (2/12)', async () => {
+    expectedText =
+      'Click on the zoom buttons to change the view of the floor plan.Alternatively, the scroll button on your mouse would work too!';
+    root.find('#tour-next-btn').first().simulate('click');
+    await wait(5);
+
+    setTimeout(() => (stepText = root.find('.reactour__helper').hostNodes().text()), 100);
+
+    expect(stepText.includes(expectedText));
+    expect(root.html()).toMatchSnapshot();
+  });
+
+  it('renders floor plan button step (3/12)', async () => {
+    expectedText =
+      'Use the floor plan button to switch between available levels and enabling / disabling the view of different components.';
```

Comment:
I'm thinking that maybe we can store the text in a Dictionary (in the tour file) and import it here.

export tourText {
step1: text,
step2: text
}

Reply:
Ohh yes ok, that would shorten the file significantly